src/test/unit/profile-upgrading.test.js

```diff
@@ -0,0 +1,285 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+// @flow
+import {
+  isOldCleopatraFormat,
+  convertOldCleopatraProfile,
+} from '../../profile-logic/old-cleopatra-profile-format';
+import {
+  isProcessedProfile,
+  upgradeProcessedProfileToCurrentVersion,
+  CURRENT_VERSION as CURRENT_PROCESSED_VERSION,
+} from '../../profile-logic/processed-profile-versioning';
+import {
+  unserializeProfileOfArbitraryFormat,
+  serializeProfile,
+} from '../../profile-logic/process-profile';
+import {
+  upgradeGeckoProfileToCurrentVersion,
+  CURRENT_VERSION as CURRENT_GECKO_VERSION,
+} from '../../profile-logic/gecko-profile-versioning';
+
+describe('upgrading old cleopatra profiles', function() {
+  const oldCleopatraProfile = require('../fixtures/upgrades/old-cleopatra-profile.sps.json');
+  const ancientCleopatraProfile = require('../fixtures/upgrades/ancient-cleopatra-profile.sps.json');
+
+  function testConvertedCleopatraProfile(profile) {
+    expect(isProcessedProfile(profile)).toBe(true);
+    // For now, just test that upgrading doesn't throw any exceptions.
+    upgradeProcessedProfileToCurrentVersion(profile);
+    expect(profile.threads.length).toBeGreaterThanOrEqual(1);
+    expect(profile.threads[0].name).toBe('GeckoMain');
+  }
+
+  it('should detect an old profile', function() {
```

Comment:
Here I removed the previous for loop, and went for a simple repetition of `it` blocks, which is more consistent with our latest practices.